sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyOptions.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.keys.cryptography;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Represents optional parameters for cryptographic operations.
+ */
+@Fluent
+public class CryptographyOptions {
+    /**
+     * Initialization vector for symmetric algorithms.
+     */
+    @JsonProperty(value = "iv")
+    private final byte[] initializationVector;
+
+    /**
+     * Additional data to authenticate but not encrypt/decrypt when using authenticated crypto algorithms.
+     */
+    @JsonProperty(value = "aad")
+    private final byte[] additionalAuthenticatedData;
+
+    /**
+     * The tag to authenticate when performing decryption with an authenticated algorithm.
+     */
+    @JsonProperty(value = "tag")
+    private final byte[] tag;
+
+    /**
+     * Creates an instance of {@link CryptographyOptions} with the given parameters.
+     *
+     * @param initializationVector Initialization vector for symmetric algorithms.
+     * @param additionalAuthenticatedData Additional data to authenticate but not encrypt/decrypt when using
+     * authenticated crypto algorithms.
+     * @param tag The tag to authenticate when performing decryption with an authenticated algorithm.
+     */
+    public CryptographyOptions(byte[] initializationVector, byte[] additionalAuthenticatedData, byte[] tag) {
```

Comment:
Is it always the case that the user must create an instance of this type by providing all three of these byte arrays? Is it not possible to provide just one or two of these? If so, having setters is more appropriate.

Reply:
Yes, see https://github.com/Azure/azure-sdk-for-net/blob/a42b92d26e9359a485fb6ec234b93146e9ff5b30/sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClient.cs#L189 for what we did for .NET after some discussion with Scott and Krzysztof.

Reply:
Changed this class to a model factory after talking with both @srnagar and @heaths.